activejob/test/cases/translation_test.rb

```diff
@@ -0,0 +1,20 @@
+require 'helper'
+require 'jobs/translated_hello_job'
+
+class TranslationTest < ActiveSupport::TestCase
+  setup do
+    JobBuffer.clear
+    I18n.available_locales = [:en, :de]
+    @job = TranslatedHelloJob.new('Johannes')
+  end
+
+  teardown do
+    I18n.available_locales = [:en]
+  end
+
+  test 'it performs the job in the given locale' do
+    @job.locale = :de
+    @job.perform_now
```

Comment:
@xijo be aware that JobBuffer won't work on the integration tests as the jobs are executed on another process. You can use the existing system in place to write content to a file(see activejob/test/support/integration/dummy_app_template.rb, activejob/test/support/integration/test_case_helpers.rb, activejob/test/integration/queuing_test.rb)